File: Quiz-server/src/main/java/com/qucat/quiz/repositories/entities/Quiz.java
package com.qucat.quiz.repositories.entities;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.qucat.quiz.repositories.entities.enums.QuizStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.sql.Timestamp;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
public class Quiz {
    private int id;
    private String name;
    private int authorId;
    private int categoryId;
    private QuizStatus status;
    private Timestamp publishedDate;
    private Timestamp updatedDate;
    private Timestamp createdDate;
    private int questionNumber;
    private int maxScore;
    private int imageId;
    private List<Question> questions;
    private Category category;
    private List<Tag> tags;
    private Image image;
    private String description;
    @JsonProperty("isFavorite")
    private boolean isFavorite;
}

File: Quiz-web/src/app/modules/core/models/quiz.ts
import {Tag} from "./tag";
import {Category} from "./category";
import {Question} from "./question";
import {Imaged} from "./imaged";
import {Image} from "./image";

export class Quiz implements Imaged {
  id: number;
  name: string;
  authorId: number;
  category: Category = new Category();
  categoryId: number;
  publishedDate: string;
  questionNumber: number;
  maxScore: number;
  tags: Tag[] = [];
  questions: Question[] = [new Question];
  imageId: number = -1;
  image: Image = new Image();
}
